java/basic-async-bot/src/main/java/com/google/chat/bot/basic/async/Bot.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.chat.bot.basic.async;
+
+// [START async-bot]
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.chat.v1.HangoutsChat;
+import com.google.api.services.chat.v1.model.Message;
+import com.google.api.services.chat.v1.model.Thread;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.logging.Logger;
+
+@SpringBootApplication
+@RestController
+public class Bot {
+  private static final Logger logger =  Logger.getLogger(Bot.class.getName());
+
+  public static void main(String[] args) {
+    SpringApplication.run(Bot.class, args);
+  }
+
+  /**
+   * Handles a GET request to the /bot endpoint.
+   *
+   * @param event Event from chat.
+   * @return Message
+   */
+  @PostMapping("/")
+  public void onEvent(@RequestBody JsonNode event) throws IOException, GeneralSecurityException {
+    String replyText = "";
+    JsonNode space = event.get("space");
+    switch (event.get("type").asText()) {
+      case "ADDED_TO_SPACE":
+        String spaceType = space.get("type").asText();
```

Comment:
can we add a private helper method that returns the value for replyText for the if/else in this case?

Reply:
Prefer not to introduce too much indirection. Normally yes, but makes it harder to follow.

That said, we should revisit these -- not sure it is necessary for every bot sample to respond to all events. The async bots are to illustrate sending the response via API instead of returning it in the response body. That can be illustrated  handling just one message type. Other ways to approach it to, but something that needs to be coordinated across the 3 languages this is implemented in.

Will file an issue to track.